### PR TITLE
Structural/add adjoint truss test

### PR DESCRIPTION
**Pull request by MFusseder**
Within this PR a new test case for the complete adjoint sensitivity analysis workflow for a linear and nonlinear truss structure is added.

### CODE REVIEW

File: applications/StructuralMechanicsApplication/tests/adjoint_sensitivity_analysis_tests/adjoint_truss_stucture_3d2n/StructuralMaterials.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
+    "properties" : [{
+        "model_part_name" : "Structure.Parts_TRUSS",
+        "properties_id"   : 1,
+        "Material"        : {
+            "constitutive_law" : {
+                "name" : "KratosMultiphysics.StructuralMechanicsApplication.TrussConstitutiveLaw"
```
Comment (armingeiser) [inline applications/StructuralMechanicsApplication/tests/adjoint_sensitivity_analysis_tests/adjoint_truss_stucture_3d2n/StructuralMaterials.json]: `KratosMultiphysics.StructuralMechanicsApplication` is not needed i think

**Review by armingeiser — CHANGES_REQUESTED** [review]
just a minor comment, otherwise good for me

File: applications/StructuralMechanicsApplication/tests/test_StructuralMechanicsApplication.py [hunk 2]
```diff
@@ -326,6 +328,8 @@ def AssembleTestSuites():
 
     smallSuite.addTests(KratosUnittest.TestLoader().loadTestsFromTestCases([TTestAdjointSensitivityAnalysisBeamStructure]))
     smallSuite.addTests(KratosUnittest.TestLoader().loadTestsFromTestCases([TTestAdjointSensitivityAnalysisShell3D3NStructure]))
+    smallSuite.addTests(KratosUnittest.TestLoader().loadTestsFromTestCases([TTestAdjointSensitivityAnalysisLinearTrussStructure]))
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/tests/test_StructuralMechanicsApplication.py]: Test are always great, but if it is a full case then consider adding it to the nightly tests

The samllsuite already takes too long (and therefore crashes Travis sometimes)

I am planning to soon move the too slow tests to the nightly suite anyway

Comment (MFusseder) [reply applications/StructuralMechanicsApplication/tests/test_StructuralMechanicsApplication.py]: I added the new cases to the nightly tests

**Review by armingeiser — APPROVED** [review]

**Pull request merged**